core/controllers/blog_dashboard.py

```diff
@@ -29,17 +29,22 @@
 import feconf
 import utils
 
+from typing import Any, Dict, List # isort:skip
 
-def _get_blog_card_summary_dicts_for_dashboard(summaries):
-    # type: (list[BlogPostSummaries]) -> list[BlogPostSummayDicts]
+
+# Here we are using Dict[str, Any] for the return value `summary_dicts` since
+# we have to return a list with each element being domain object converted to
+# a dictionary.
+def _get_blog_card_summary_dicts_for_dashboard(
```

Comment:
Thanks for fixing this. I'm somewhat concerned that this wasn't caught by mypy before...any idea why?

(This is serious because it suggests that future errors won't be caught either.)

Reply:
cc @hardikkat24

Reply:
Found the reason.
This file is in the denylist. See [here](https://github.com/oppia/oppia/blob/bf4d1e20653c6fba39445e2cdd073071cc866316/scripts/run_mypy_checks.py#L52).

Reply:
I am keeping the file in denylist only - reason whenever this file is fully type annotated, it will be to safe to remove it from denylist then. But I have decided to keep the rest of the annotations in the file. Added explanation wherever Any is used here. See the error log below if I remove the file from denylist, which shows it is not yet ready:
![Screenshot from 2021-08-17 15-04-45](https://user-images.githubusercontent.com/32328861/129702122-4ff79674-667f-4f21-9ead-d0e41940aac3.png)